### PR TITLE
Model

### PR DESCRIPTION
**Pull request by elilionel7**
This PR introduces the PetAdoptionTrainer class for training and evaluating pet adoption prediction models. Key features include:

Integration of XGBoost for robust model training.
Data preprocessing and handling of class imbalances with SMOTE.
Support for both standard training and cross-validation.
Detailed evaluation metrics, including F1 Score, Accuracy, and Recall.
Functionality for saving the trained model.

### CODE REVIEW

File: src/model_scripts/train_data.py [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+        y = self.preprocessed_data["Adopted"]
+        X_temp, self.X_test, y_temp, self.y_test = train_test_split(
+            X, y, test_size=test_size, random_state=42
+        )
+        self.X_train, self.X_val, self.y_train, self.y_val = train_test_split(
+            X_temp, y_temp, test_size=validation_size, random_state=42
+        )
+
+    def train_model(self, use_cv=False, cv_folds=5):
+        # """
```
Comment (theSekyi) [inline src/model_scripts/train_data.py]: Make the docstrings consistent

Comment (elilionel7) [reply src/model_scripts/train_data.py]: Yeah…
